### DISCUSSION TITLE
WHY PLOTER SHOWS 2 CHANNEL DATA

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#938] [Q&A] [answered]
below is code were i  am  convert  i2s to single channel data then why is plotter  showing 2 channel data ?I  didn't fill both channel .  I was just testing the code and functionality

IN MY CASE It should be single channel data while the second channel should empty .
 
#include "AudioTools.h"
#include "AudioLibs/AudioA2DP.h"
CsvOutput<int16_t> out(Serial, 2); // ASCII stream as csv 

I2SStream i2sStream;                            // Access I2S as stream
 
 
FormatConverterStream conv(out);
 

StreamCopy copierr(conv, i2sStream);
 
void setup(void) {
    Serial.begin(115200);
 
   
  auto config_in = i2sStream.defaultConfig(RX_MODE);
  config_in.i2s_format = I2S_STD_FORMAT; // or try with I2S_LSB_FORMAT | I2S_STD_FORMAT
  config_in.is_master = true;
  config_in.use_apll = false;  // try with yes
  config_in.pin_mck = 3;
  config_in.bits_per_sample = 32; 
  config_in.channels = 2; 
  config_in.sample_rate = 44100; 
  i2sStream.begin(config_in);
  AudioInfo from(44100, 2, 32);
  AudioInfo to  (44100, 1, 16);

  conv.begin(from, to);

}
void loop() {
  
    copierr.copy( );
 
}

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/a8077baf-2f3b-4fdc-bf36-cabdaff8c7b7)

**Comment by pschatzmann** [comment, marked as answer]
Because you defined it to be 2 channels in the constructor
